### ISSUE TITLE
Timeouts With Many Jobs

### ISSUE THREAD

**Issue by jakemumu** [kenaniah/sidekiq-status#42]
Hey! Loving this gem, but when I run a large batch of like 100k users, the statuses tab times out in my app, is there a fix for that somehow?

**Comment by Gooner91** [comment]
Yep, can confirm, the show view for individual jobs work but the index view results in gateway timeout.

**Comment by Andy9822** [comment]
I came across this issue recently and the gem was unusable in our Production environment due to the large number of jobs present in the Redis.

I quickly looked at the source code and suspected the way the Redis keys were accessed and processed was super inefficient. I tried a very simple alternative and this was my result:

P.S. `zoo` was the method using the gem's code for doing the Redis keys manipulation while `foo` was my fix attempt - which turned out to be 9x faster.
P.S.2 Ignore the `jid` logs, it was leftovers from my first test attempts
> ![image (8)](https://github.com/kenaniah/sidekiq-status/assets/19315023/97d613ef-813b-42f1-96ca-4f3271bb030e)


I don't know how maintained this gem currently is - I just left an issue [here](https://github.com/kenaniah/sidekiq-status/issues/46). Depending on it, I may take some time to give my idea a shot and open a PR to improve the performance so we avoid the timeout - or at least increase the size of jobs that the gem can process without causing a timeout
